platform/src/main/java/org/hillview/sketches/Histogram.java

```diff
@@ -52,6 +64,10 @@ public void add(IColumn column, int currRow) {
             int index = this.bucketDescription.indexOf(column, currRow);
             if (index >= 0)
                 this.buckets[index]++;
+
+            for (int i = 0; i < index+1; i++) {
```

Comment:
Hmm. This was intended for backwards-compatibility so that the CDF for the partial result would be sensible, and then the front-end doesn't have to have two separate procedures for computing the CDF (integrating the histogram vs. using the CDF from the backend). If it's computed at the root, the partial rendering won't work anymore, I think. But it might be expensive, yes.